frontend-ssr/pages/submissions.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import Head from 'next/head';
+import Footer from '~/components/footer/Footer.component';
+import Header from '~/components/header/Header.component';
+
+import ExerciseSubmission from '../components/exercise-submissions/ExerciseSubmission';
+import ExerciseService from '../services/ExerciseSubmissions.service';
+import {
+  loadSubmissions,
+  searchSubmissions,
+} from '~/redux/exercise-submissions/exercise-submissions.actions';
+import ExerciseSubmissionSkeleton from '~/components/exercise-submissions/ExerciseSubmissionSkeleton';
+import SweetAlertService from '../services/sweet-alert/SweetAlert.service';
+import Search from '../components/Search';
+import { RootState } from '../redux/root.reducer';
+
+import styles from './submissions.module.scss';
+
+interface State {
+  loading: boolean;
+}
+
+const mapState = (state: RootState) => ({
+  submissions: state.submissions,
+});
+
+class Submissions extends React.Component<ConnectedProps<typeof connector>, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      loading: props.submissions.submissions === undefined,
+    };
+  }
+
+  componentDidMount() {
+    let { submissions } = this.props;
+
+    if (submissions.submissions === undefined) {
+      this.getData();
+    }
+  }
+
+  getData = async () => {
+    const { submissions, dispatch } = this.props;
+
+    try {
+      this.setState({
+        loading: true,
+      });
+      const response = await ExerciseService.getSubmissions(
+        submissions.page,
+        '',
+      );
+      const jsonResp = await response.json();
+      this.setState({
+        loading: false,
+      });
+      dispatch(loadSubmissions(jsonResp));
+    } catch (err) {
+      SweetAlertService.toast({ type: 'error', text: err });
+      this.setState({
+        loading: false,
+      });
+    }
+  };
+
+  searchData = async (e: any) => {
+    e.preventDefault();
```

Comment:
This prevent default should be in `Search` component. Otherwise we'll have to add `e.preventDefault` in every component that uses search. 

Also, the search should call this function directly with the query string as the parameter. So that here we don't have to do `e.target.query.value`.